core/trino-main/src/main/java/io/trino/sql/ir/optimizer/rule/SimplifyRedundantCase.java

```diff
@@ -53,11 +54,11 @@ public Optional<Expression> apply(Expression expression, Session session, Map<Sy
 
         WhenClause thenClause = caseTerm.whenClauses().getFirst();
         if (thenClause.getResult().equals(TRUE) && caseTerm.defaultValue().equals(FALSE)) {
-            return Optional.of(thenClause.getOperand());
+            return Optional.of(new Comparison(Comparison.Operator.IDENTICAL, thenClause.getOperand(), TRUE));
```

Comment:
Why not `coalesce(thenClause.getOperand(), false)`?

This would let coalesce pruning to kick in and simplify this further.

Reply:
Good question!

That's a larger topic of discussion, but the short answer is that coalesce is a conditional expression and it makes it harder, in general, for the engine to reason about. It turns out that using `x ≡ true` and `x  ≡ false` is a much better way to turn ternary logic into binary logic, do transformations that preserve properties, etc.

Reply:
do we have a rule optimizing `x ≡ true` to `x` when `x` is not nullable?

Reply:
Not yet, because we don't have a way to know if x is nullable or not. But that's coming, too.